lib/babel/transformation/transformers/es6/parameters.rest.js

```diff
@@ -19,6 +77,24 @@ exports.Function = function (node, parent, scope) {
   // otherwise `arguments` will be remapped in arrow functions
   argsId._ignoreAliasFunctions = true;
 
+  // check if rest is used only in member expressions
+  var restOuterDeclar = scope.getBindingIdentifier(rest.name);
+  var state = {
+    name: rest.name,
+    outerDeclar: restOuterDeclar,
+    isOptimizable: true,
+    candidates: []
+  };
+  scope.traverse(node, memberExpressionVisitor, state);
+
+  if (state.isOptimizable) {
```

Comment:
Well, I do not think that these could be moved into memberExpressionVisitor without breaking everything. For example, I have to compare binding identifiers to check for this:

    function j(foo, ...bar) {
      var x = function z(bar) {
        bar[1] = 5;
      };
    }

Otherwise it would trigger array copying.